# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsWindowWatcher.cpp
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 toolkit/components/windowwatcher/nsWindowWatcher.cpp
--- a/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Tue Apr 07 15:17:14 2020 +0000
+++ b/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Tue Apr 07 15:17:47 2020 +0000
@@ -68,7 +68,6 @@
 #include "mozilla/dom/BrowserParent.h"
 #include "mozilla/dom/BrowserHost.h"
 #include "mozilla/dom/DocGroup.h"
-#include "mozilla/dom/TabGroup.h"
 #include "nsIAppWindow.h"
 #include "nsIXULBrowserWindow.h"
 #include "nsGlobalWindow.h"
@@ -986,10 +985,10 @@
       } else {
         newBC->SetOpener(parentBC);
       }
-    } else if (parentWindow && parentWindow != win) {
-      MOZ_ASSERT(
-          win->TabGroup() != parentWindow->TabGroup(),
-          "If we're forcing no opener, they should be in different tab groups");
+    } else if (parentBC && parentBC != newBC && !newBC->IsChrome()) {
+      MOZ_ASSERT(newBC->Group() != parentBC->Group(),
+                 "If we're forcing no opener, they should be in different "
+                 "browsing context groups");
     }
 
     if (windowIsNew) {